File: easy/3880_minimum_absolute_difference_between_two_values/solution.cpp
#include <vector>

class Solution {
public:
  int minAbsoluteDifference(std::vector<int> &nums) {
    int minDifference = INT_MAX;

    int prevNum = 0;
    int prevI = 0;

    for (int i = 0; i < nums.size(); ++i) {
      if (nums[i] == 0)
        continue;

      if (prevNum == 3 - nums[i] && i - prevI < minDifference)
        minDifference = i - prevI;

      prevNum = nums[i];
      prevI = i;
    }

    return minDifference != INT_MAX ? minDifference : -1;
  }
};

File: easy/3880_minimum_absolute_difference_between_two_values/solution.py
class Solution:
    def minAbsoluteDifference(self, nums: list[int]) -> int:
        min_difference = float("inf")

        prev_num = 0
        prev_i = 0

        for i, num in enumerate(nums):
            if num == 0:
                continue

            if prev_num == 3 - num and i - prev_i < min_difference:
                min_difference = i - prev_i

            prev_num = num
            prev_i = i

        return min_difference if min_difference != float("inf") else -1
